src/audio_core/delay_line.cpp

```diff
@@ -0,0 +1,103 @@
+#include "audio_core/delay_line.h"
+
+namespace AudioCore {
+DelayLineBase::DelayLineBase() = default;
+DelayLineBase::~DelayLineBase() = default;
+
+void DelayLineBase::Initialize(s32 _max_delay, float* src_buffer) {
+    buffer = src_buffer;
+    buffer_end = buffer + _max_delay;
+    max_delay = _max_delay;
+    output = buffer;
+    SetDelay(_max_delay);
+    Clear();
+}
+
+void DelayLineBase::SetDelay(s32 new_delay) {
+    if (max_delay < new_delay) {
+        return;
+    }
+    delay = new_delay;
+    input = (buffer + ((output - buffer) + new_delay) % (max_delay + 1));
+}
+
+s32 DelayLineBase::GetDelay() const {
+    return delay;
+}
+
+s32 DelayLineBase::GetMaxDelay() const {
+    return max_delay;
+}
+
+f32 DelayLineBase::TapOut(s32 last_sample) {
+    float* ptr = input - (last_sample + 1);
```

Comment:
```suggestion
    const float* ptr = input - (last_sample + 1);
```

Would allow this function to be turned into a const qualified one.